src/library_html5.js

```diff
@@ -1557,7 +1557,7 @@ var LibraryJSEvents = {
   },
 
   emscripten_set_pointerlockchange_callback: function(target, userData, useCapture, callbackfunc) {
-    if (!document.body.requestPointerLock && !document.body.mozRequestPointerLock && !document.body.webkitRequestPointerLock && !document.body.msRequestPointerLock) {
+    if (!document.body || (!document.body.requestPointerLock && !document.body.mozRequestPointerLock && !document.body.webkitRequestPointerLock && !document.body.msRequestPointerLock)) {
```

Comment:
Oh, this is not only for `--proxy-to-worker`, but also if pthreads happen to call these functions. Those don't get the proxy-to-worker shims.

Although, I'm not a big fan of polyfilling the worker environment in proxy-to-worker to mimic to look like the main thread, because it might be confusing to outside code, which might be looking at the presence of `document.body` or other types of global vars to behave in some kind of fashion that'd expect full support for the feature.